### PR TITLE
Add recipe for grenedalf

### PR DESCRIPTION
**Pull request by lczech**
Add recipe for grenedalf, https://github.com/lczech/grenedalf

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by lczech** [comment]
Dear @bioconda/core @bioconda/<team>,

could you please have a look at the failing check here? The error is a rather unspecific:

>    BIOCONDA ERROR COMMAND FAILED (exited with 1): /opt/mambaforge/envs/bioconda/bin/conda-build --override-channels --no-anaconda-upload -c conda-forge -c bioconda -e /opt/mambaforge/envs/bioconda/conda_build_config.yaml -e /opt/mambaforge/envs/bioconda/lib/python3.10/site-packages/bioconda_utils/bioconda_utils-conda_build_config.yaml recipes/grenedalf/meta.yaml

without further information (that I could find) on what exactly the issue is. Hence, I have no idea how to fix this.

Thanks in advance
Lucas

PS: In the above PR template, it says to write to `@bioconda/<team>`, but I could not find any information on what to replace `<team>` with - I hope the above `core` is fine, as that's the only one that is also mentioned there.

File: recipes/grenedalf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  url: https://github.com/lczech/grenedalf/archive/refs/tags/v0.5.2.tar.gz
+  sha256: 03d5391be83532f67b4aaedeeb7275cdf1de19070096a23c48367b7cea902b4e
+
+requirements:
+  build:
+    - make
+    - cmake
+    - autoconf
+    - automake
+    - {{ compiler('cxx') }}
```
Comment (martin-g) [inline recipes/grenedalf/meta.yaml:20]:
```suggestion
    - {{ compiler('cxx') }}
    - libcxx < 17 # [osx]
```

File: recipes/grenedalf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  host:
+    - bzip2
+    - xz
+    - zlib
```
Comment (martin-g) [inline recipes/grenedalf/meta.yaml:24]:
```suggestion
    - zlib
    - libcxx < 17 # [osx]
```

File: recipes/grenedalf/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    grenedalf is a collection of commands for working with population genetic data, in particular from pool sequencing.
+    Its main focus are statistical analyses such as Tajima's D and Fst. The statistics follow the approaches of PoPoolation
+    and PoPoolation2, as well as poolfstat and npstat. However, compared to those, grenedalf is significantly more scalable,
+    more user friendly, and offers more settings and input file formats.
+  dev_url: https://github.com/lczech/grenedalf
+  doc_url: https://github.com/lczech/grenedalf/wiki
+
+extra:
+  additional-platforms:
+    - linux-aarch64
```
Comment (martin-g) [inline recipes/grenedalf/meta.yaml:49]:
```suggestion
    - linux-aarch64
    - osx-arm64
```

File: recipes/grenedalf/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: 03d5391be83532f67b4aaedeeb7275cdf1de19070096a23c48367b7cea902b4e
+
+requirements:
+  build:
+    - make
+    - cmake
+    - autoconf
+    - automake
+    - {{ compiler('cxx') }}
+    - libcxx < 17 # [osx]
```
Comment (martin-g) [inline recipes/grenedalf/meta.yaml:21]:
```suggestion
    - libcxx <17 # [osx]
```

File: recipes/grenedalf/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  host:
+    - bzip2
+    - xz
+    - zlib
+    - libcxx < 17 # [osx]
```
Comment (martin-g) [inline recipes/grenedalf/meta.yaml:26]:
```suggestion
    - libcxx <17 # [osx]
```

**Comment by lczech** [comment]
Hey @martin-g,

thank you very much for suggesting the fixes by specifying different clang versions for OSX! That seems to be the way, and has partially worked, but not for osx-arm64, it seems. In my CI tests, I did have some trouble getting Clang 16 to work on osx-arm64, which is what the current recipe is using. It might be that the issue here is similar, and hence I tried to get the recipe to use Clang 15 on osx-arm64 instead. However, so far I could not get it to work, with conda reporting some incompatibilities of dependencies. 

You are more experienced with these things that I am - I'd greatly appreciate if you could give it a shot, and change the recipe to use Clang 15 on osx-arm64. I think that might work. Alternatively, it would also be okay to exclude osx-arm64 for now, to get the tool out there at least.

Thanks a lot in advance
Lucas

**Comment by martin-g** [comment]
Please add a skip for osx and remove all deps related to osx. 
Later when someone needs OSX build could try to debug the issues!

File: recipes/grenedalf/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
+package:
+  name: grenedalf
+  version: "0.5.2"
+
+build:
+  number: 0
+  skip: {{ os.environ['TRAVIS_OS_NAME'] == 'osx' }}
```
Comment (martin-g) [inline recipes/grenedalf/meta.yaml:7]: Travis is not used here at all. 
```suggestion
  skip: true # [osx]
```
I think you can add back linux-aarch64

Comment (lczech) [reply recipes/grenedalf/meta.yaml]: Ah, thank you, that was what I was looking for! Are these things documented anywhere? I've failed finding information on this :-(

**Comment by lczech** [comment]
Okay, good enough for now - seems to work on Linux. MacOS later. Thanks again @martin-g for your help!

**Comment by lczech** [comment]
@BiocondaBot please add label

File: recipes/grenedalf/meta.yaml (hunk 6 above)
Comment (martin-g) [reply recipes/grenedalf/meta.yaml]: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#skipping-builds
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocess-selectors

**Review by martin-g — APPROVED** [review]

**Pull request merged**